gdal/ogr/ogrct.cpp

```diff
@@ -586,8 +582,11 @@ class OGRProjCT : public OGRCoordinateTransformation
             minx(minxIn), miny(minyIn), maxx(maxxIn), maxy(maxyIn),
             pj(pjIn), osName(osNameIn), osProjString(osProjStringIn),
             accuracy(accuracyIn) {}
-
-        Transformation(const Transformation&) = delete;
+        Transformation(const Transformation& other):
+            minx(other.minx), miny(other.miny), maxx(other.maxx), maxy(other.maxy),
+            pj((other.pj != nullptr) ? (proj_clone(OSRGetProjTLSContext(), other.pj)) : (nullptr)),
```

Comment:
Perhaps a todo for PROJ to return NULL when you clone a NULL?  Or even better, have a copy ctor? and then this can all go away?

Reply:
This almost argues for wrapping the PJ so that you can hide the proj badness there and then this can all vanish.

Reply:
`proj_clone` return `nullptr` for null object (see https://github.com/OSGeo/PROJ/blob/master/src/iso19111/c_api.cpp#L510), but set error. I'm think it is not needed in this case since null `other.pj` can be valid value.

Reply:
My point was that if the pj object was simply wrapped, you could do whatever you wanted in that wrapped structure's copy ctor and all of these copy ctors can just be made default. The only reason that you need this copy ctor is because you have no good copy ctor for PJ.

Reply:
Take a [look](https://github.com/OSGeo/gdal/pull/3809/commits/82b56872fa0c70d7b167efbef91a1fed8cfd20bb).
Not sure if it got better.

Reply:
Maybe I missed that you have to also deal with the OGR SRS src/target on line 977?
I was hoping it would be smaller. But it's contained and perhaps usable in GDAL projections, too? Maybe someone else has an opinion.